src/components/20-molecules/image-upload/ui.test.js

```diff
@@ -0,0 +1,218 @@
+    () => document.querySelector('axa-image-upload').shadowRoot
+  );
+  const $axaElemShadowEl = await $axaElemShadow.find(IMAGE_UPLOAD_CLASS);
+  await t.expect($axaElemShadowEl.exists).ok();
+});
+
+test('should upload file and switch view', async t => {
+  const $inputFileInputElem = await Selector(
+    () =>
+      document
+        .querySelector('axa-image-upload')
+        .shadowRoot.querySelector('axa-input-file .a-input-file__input') // TODO change to .js-
+  );
+  await t.expect($inputFileInputElem.exists).ok();
+
+  await t.setFilesToUpload($inputFileInputElem, validFiles);
+
+  const $fileOverviewElem = await Selector(() =>
+    document
+      .querySelector('axa-image-upload')
+      .shadowRoot.querySelector('.js-image-upload__dropzone-file-overview')
+  );
+  await t.expect($fileOverviewElem.exists).ok();
+});
+
+test("shouldn't upload file because its an invalid type", async t => {
+  const $inputFileInputElem = await Selector(() =>
+    document
+      .querySelector('axa-image-upload')
+      .shadowRoot.querySelector('axa-input-file .a-input-file__input')
+  );
+  await t.expect($inputFileInputElem.exists).ok();
+
+  await t.setFilesToUpload($inputFileInputElem, './ui-test-files/test.rtf');
+
+  const $fileOverviewElem = await Selector(() =>
+    document
+      .querySelector('axa-image-upload')
+      .shadowRoot.querySelector('.js-image-upload__dropzone-file-overview')
+  );
+  await t.expect($fileOverviewElem.exists).notOk();
+});
+
+test('should cobvert .png file to .jpg', async t => {
+  const $inputFileInputElem = await Selector(() =>
+    document
+      .querySelector('axa-image-upload')
+      .shadowRoot.querySelector('axa-input-file .a-input-file__input')
+  );
+  await t.expect($inputFileInputElem.exists).ok();
+
+  await t.setFilesToUpload($inputFileInputElem, validFiles[1]); // files[1] is a .png file
+
+  const $figcaptionElem = await Selector(() =>
+    document
+      .querySelector('axa-image-upload')
+      .shadowRoot.querySelector(
+        '.js-image-upload__img-figure .js-image-upload__img-caption'
+      )
+  );
+  await t
+    .expect((await $figcaptionElem.getAttribute('title')) === 'test.jpg')
+    .ok();
+});
+
+test('should delete image', async t => {
+  const $inputFileInputElem = await Selector(() =>
+    document
+      .querySelector('axa-image-upload')
+      .shadowRoot.querySelector('axa-input-file .a-input-file__input')
+  );
+  await t.expect($inputFileInputElem.exists).ok();
+
+  await t.setFilesToUpload($inputFileInputElem, validFiles[1]);
+
+  const $figureElem = await Selector(() =>
+    document
+      .querySelector('axa-image-upload')
+      .shadowRoot.querySelector('.js-image-upload__img-figure')
+  );
+
+  const $figureElemsbBefore = await Selector(() =>
+    document
+      .querySelector('axa-image-upload')
+      .shadowRoot.querySelectorAll('.js-image-upload__img-figure')
+  );
+
+  await t.expect($figureElemsbBefore.count).eql(2); // one file + addMoreInputFile
```

Comment:
`figureElementsBefore `

Reply:
called it `figureElemsBefore`cause other variable is called `figureElemsAfter`

Reply:
That's ok, but there is a typo in here. Double `b`